packages/core/src/components/DependencyGraph/DependencyGraph.tsx

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import * as d3Zoom from 'd3-zoom';
+import * as d3Selection from 'd3-selection';
+import useTheme from '@material-ui/core/styles/useTheme';
+import dagre from 'dagre';
+import debounce from 'lodash/debounce';
+import { BackstageTheme } from '@backstage/theme';
+import Node from './Node';
+import {
+  DependencyEdge,
+  DependencyNode,
+  Direction,
+  Alignment,
+  Ranker,
+  RenderNodeFunction,
+  RenderLabelFunction,
+  GraphEdge,
+  GraphNode,
+  LabelPosition,
+} from './types';
+import Edge from './Edge';
+import { ARROW_MARKER_ID } from './constants';
+
+export type DependencyGraphProps = React.SVGProps<SVGSVGElement> & {
+  edges: DependencyEdge[];
+  nodes: DependencyNode[];
+  direction?: Direction;
+  align?: Alignment;
+  nodeMargin?: number;
+  edgeMargin?: number;
+  rankMargin?: number;
+  marginX?: number;
+  marginY?: number;
```

Comment:
Hm, is this an inner margin that effectively acts as a padding from the outside point of view?

Reply:
Yeah, that's true. I just directly used the options from `dagre`, padding would make more sense to a user. Will rename!